### ISSUE TITLE
delete and append to Tfrecords: Tensorflow

### ISSUE THREAD

**Issue by samra-irshad** [tensorflow/tensorflow#31738]
I am currently working on a research in which I have to append my training set after some epochs and delete some samples from test set after evaluation. Currently there is no way with which I can access the records (placed at specific indexes) in tfrecord file. Since tfrecords offer very fast training so I avoid using generators. Any chance of adding such feature?

**Comment by gadagashwini** [comment]
@samra-irshad, Will it be possible to provide the sample code and also include the tensorflow version. Thanks!

**Comment by samra-irshad** [comment]
@gadagashwini My post is a feature request related to tfrecords in tensorflow.

**Comment by samra-irshad** [comment]
I can explain what I am doing:
1. Train a model on training data with tf.dataset 
2. Evaluate the model on test data.
3. Now in next phase, I need to remove some samples from test set and add some samples in training set. So I have to iteratively create tfrecord files, which I am doing currently. But the problem is it takes a very long time to create tfrecord files (in middle of training) repeatedly in middle of training

I believe it would be great if we can have an indexing mechanism for tfrecord files
